plutus-tx/test/Plugin/Spec.hs

```diff
@@ -392,10 +403,157 @@ evenMutual = plc @"evenMutual" (
         odd n = if n == 0 then False else even (n-1)
     in even)
 
+readCompiledCode :: PLC.KnownDynamicBuiltinType a => (a -> b) -> CompiledCode b -> b
+readCompiledCode inj compiled =
+    case PLC.readDynamicBuiltinCek term of
+        Right (Just x) -> inj x
+        Left _         -> error "Can't read compiled code"
+  where
+    PLC.Program _ _ term = getPlc compiled
+
+readMyMonoRecord :: CompiledCode MyMonoRecord -> MyMonoRecord
+readMyMonoRecord = readCompiledCode (uncurry MyMonoRecord)
+
+myMonoRecord01 :: MyMonoRecord
+myMonoRecord01 = MyMonoRecord 0 1
+
+compiledMyMonoRecord01 :: CompiledCode MyMonoRecord
+compiledMyMonoRecord01 = plc @"compiledMyMonoRecord01" (MyMonoRecord 0 1)
+
+readMyMonoRecords :: CompiledCode [MyMonoRecord] -> [MyMonoRecord]
+readMyMonoRecords = readCompiledCode (Prelude.map (uncurry MyMonoRecord) . PLC.unPlcList)
+
+myMonoRecords :: [MyMonoRecord]
+myMonoRecords = Prelude.zipWith MyMonoRecord [0..4] [1..5]
+
+compiledMyMonoRecords :: CompiledCode [MyMonoRecord]
+compiledMyMonoRecords = plc @"compiledMyMonoRecords" (
+        [MyMonoRecord 0 1, MyMonoRecord 1 2, MyMonoRecord 2 3, MyMonoRecord 3 4, MyMonoRecord 4 5]
+    )
+
+-- GHC can reorder constructors, but here we hardcode a particular order.
```

Comment:
I don't understand what this is referring to. We fix the order that we consider constructors in, see the "Ordering of constructors" note.

Reply:
Ah, thanks, I'll remove that.